src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/MetadataReader.netstandard.cs

```diff
@@ -101,6 +108,67 @@ public static unsafe AssemblyName GetAssemblyName(string assemblyFile)
             }
         }
 
+        private static void GetPEKind(PEReader peReader, out PortableExecutableKinds peKind, out ImageFileMachine machine)
+        {
+            PEHeaders peHeaders = peReader.PEHeaders;
+            PEMagic peMagic = peHeaders.PEHeader!.Magic;
+            Machine coffMachine = peHeaders.CoffHeader.Machine;
+            CorFlags corFlags = peHeaders.CorHeader!.Flags;
+
+            peKind = default;
+            if ((corFlags & CorFlags.ILOnly) != 0)
+                peKind |= PortableExecutableKinds.ILOnly;
+
+            if ((corFlags & CorFlags.Prefers32Bit) != 0)
+                peKind |= PortableExecutableKinds.Preferred32Bit;
+            else if ((corFlags & CorFlags.Requires32Bit) != 0)
+                peKind |= PortableExecutableKinds.Required32Bit;
+
+            if (peMagic == PEMagic.PE32Plus)
+                peKind |= PortableExecutableKinds.PE32Plus;
+
+            machine = (ImageFileMachine)coffMachine;
+        }
+
+        private static ProcessorArchitecture CalculateProcArchIndex(PortableExecutableKinds pek, ImageFileMachine ifm, AssemblyFlags flags)
```

Comment:
I wonder if this could be shared with https://github.com/dotnet/runtime/blob/6d0bcc4cc7cf98e661c91d4f2abace2c5bd282a5/src/coreclr/System.Private.CoreLib/src/System/Reflection/AssemblyName.CoreCLR.cs#L113

I could not figure what could be a good place to put the shared implementation or whether it is worth the trouble.